### PR TITLE
Fix API Module

### PR DESCRIPTION
**Pull request by ty-everett**
Fix all errors preventing the API module (server backend) from working correctly.

### CODE REVIEW

**Comment by ty-everett** [comment]
There are still errors building the API. I can't seem to figure out why rollup isn't building the inject script. Once the API is working correctly and the inject script builds (to gateway.cash/pay.js), we will be able to bring gateway.cash back online under the new infrastructure.
Any help debugging or solving the build errors is much appreciated.

**Comment by ty-everett** [comment]
Resolved the error with rollup. Correct me if wrong but I think the format of the PayButton should be "esm" (output.format in rollup.config.js), rather than "cjs". I am otherwise unable to import it using ES6 syntax and am forces to use the CommonJS require(...). If someone knows a way to allow it to be imported using both methods (ES6 and CJS), let me know so we can remain as compatible as possible.

If we are only able to use one, my vote is for "esm" format since you generally use "import PayButton from '@gatewaycash/paybutton'" rather than "require('@gatewaycash/paybutton')" when developing with React.

Thoughts welcome, not merging this yet since API is still broken and dotenv, setup scripts need setting up for frontend and PayButton.

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,9 @@ node_modules
 
 # parcel-bundler cache (https://parceljs.org/)
 .cache
+
+# ignore backend sessions (and Jeff Sessions?)
```
Comment (alex-wilmer) [inline .gitignore]: lol

**Comment by ty-everett** [comment]
I've finished the registration endpoint and others will follow over time. This PR has gotten far too large and so I am merging it now in order to keep master up-to-date and to allow other things to get worked on. We are moving away from sessions in favor of API keys but staying with MySQL for the present moment since there isn't a specific need to migrate yet.

**Pull request merged**